stdlib/public/core/Span/Span.swift

```diff
@@ -0,0 +1,827 @@
+  ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  @lifetime(immortal)
+  public init(
+    _unsafeBytes buffer: Slice<UnsafeMutableRawBufferPointer>
+  ) {
+    self.init(_unsafeBytes: UnsafeRawBufferPointer(rebasing: buffer))
+  }
+}
+
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 9999, *)
+extension Span where Element: Equatable {
+
+  /// Returns a Boolean value indicating whether this and another span
+  /// contain equal elements in the same order.
+  ///
+  /// - Parameters:
+  ///   - other: A span to compare to this one.
+  /// - Returns: `true` if this sequence and `other` contain equivalent items,
+  ///   using `areEquivalent` as the equivalence test; otherwise, `false.`
+  ///
+  /// - Complexity: O(*m*), where *m* is the lesser of the length of the
+  ///   sequence and the length of `other`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _elementsEqual(_ other: Self) -> Bool {
+    guard count == other.count else { return false }
+    if count == 0 { return true }
+
+    //FIXME: This could be short-cut
+    //       with a layout constraint where stride equals size,
+    //       as long as there is at most 1 unused bit pattern.
+    // if Element is BitwiseEquatable {
+    // return _swift_stdlib_memcmp(lhs.baseAddress, rhs.baseAddress, count) == 0
+    // }
+    for o in 0..<count {
+      if self[unchecked: o] != other[unchecked: o] { return false }
+    }
+    return true
+  }
+
+  /// Returns a Boolean value indicating whether this span and a Collection
+  /// contain equal elements in the same order.
+  ///
+  /// - Parameters:
+  ///   - other: A Collection to compare to this span.
+  /// - Returns: `true` if this sequence and `other` contain equivalent items,
+  ///   using `areEquivalent` as the equivalence test; otherwise, `false.`
+  ///
+  /// - Complexity: O(*m*), where *m* is the lesser of the length of the
+  ///   sequence and the length of `other`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _elementsEqual(_ other: some Collection<Element>) -> Bool {
+    guard count == other.count else { return false }
+    if count == 0 { return true }
+
+    return _elementsEqual(AnySequence(other))
+  }
+
+  /// Returns a Boolean value indicating whether this span and a Sequence
+  /// contain equal elements in the same order.
+  ///
+  /// - Parameters:
+  ///   - other: A Sequence to compare to this span.
+  /// - Returns: `true` if this sequence and `other` contain equivalent items,
+  ///   using `areEquivalent` as the equivalence test; otherwise, `false.`
+  ///
+  /// - Complexity: O(*m*), where *m* is the lesser of the length of the
+  ///   sequence and the length of `other`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _elementsEqual(_ other: some Sequence<Element>) -> Bool {
```

Comment:
How is the ambiguity between Collection and Sequence resolved? Does it always resolve to Collection because Collection extends Sequence and is therefore “more specific”?

Reply:
Yes; if a type conforms to `Collection` that overload will be chosen as more specific.